src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/bot_publish_prep.py

```diff
@@ -64,7 +64,7 @@ def prepare_publish_v4(logger, code_dir, proj_file):
                 logger.info('Detected bot language C#, Bot Builder version v4.')
 
                 if proj_file is None:
-                    raise CLIError('Expected --proj-file argument provided with the full path to the '
+                    raise CLIError('Expected --proj-name argument provided with the full path to the '
```

Comment:
So then why does the help say it is just the name of the file?  It should say it is the path. Also, calling the argument `--proj-name` and expecting a path doesn't make sense.

Reply:
Unfortunately that argument name is what we landed on in the first refactor back in Nov 2018; I agree that it is poorly named. 

We don't want to break current users of the CLI, is there a way we can deprecate this arg? Could we set in the _params.py for the possible flags that it uses `['--proj-name', '--proj-file']` (or `--proj-file-path`) and start directing users to use `--proj-file` or is that considered worse form?

Reply:
You could have two options, with `--proj-name` deprecated and hidden. 
https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#deprecating-commands-and-arguments

Scroll down to "deprecation argument option"